sceptre/resolvers/__init__.py

```diff
@@ -49,62 +49,231 @@ def resolve(self):
         """
         pass  # pragma: no cover
 
+    def clone(self, stack: 'stack.Stack') -> 'Resolver':
+        """
+        Produces a "fresh" copy of the Resolver, with the specified stack.
+
+        :param stack: The stack to set on the cloned resolver
+        """
+        return type(self)(self.argument, stack)
+
 
-class ResolvableProperty(object):
+class ResolvableProperty(abc.ABC):
```

Comment:
I've made ResolvableProperty into a base class with (currently) one implementation, ResolvableContainerProperty. This is necessary so in my next PR I can make the ResolvableValueProperty.